### PR TITLE
Picture cache

### PR DESCRIPTION
**Pull request by RootofalleviI**

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java [hunk 1]
```diff
@@ -132,34 +135,39 @@ private void setupFavorListener(View rootView, String favorId) {
...
     if (url != null) {
-      v.findViewById(R.id.loading_panel).setVisibility(View.VISIBLE);
-      getViewModel()
-          .downloadPicture(currentFavor)
-          .thenAccept(
-              picture -> {
-                mImageView.setImageBitmap(picture);
-                v.findViewById(R.id.loading_panel).setVisibility(View.GONE);
-              });
+      Bitmap pic = getViewModel().loadPictureFromLocal(getContext(), currentFavor).get();
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java:151]: I think this logic should be placed in the FavorViewModel directly and wrapped inside a single method that returns a picture.

File: app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java [hunk 2]
```diff
@@ -1,253 +1,282 @@
...
+
+  // Upload/download pictures
+  @Override
+  public void uploadOrUpdatePicture(Favor favor, Bitmap picture) {
+    CompletableFuture<String> pictureUrl = getPictureUtility().uploadPicture(picture);
+    pictureUrl.thenAccept(url -> getFavorRepository().updateFavorPhoto(favor, url));
+  } // check what happens if updateFavorFoto fails
+
+  @Override
+  public CompletableFuture<Bitmap> downloadPicture(Favor favor) throws RuntimeException {
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:163]: Should probably make this private

File: app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java [hunk 3]
```diff
@@ -1,253 +1,282 @@
...
+        }
+      };
+    } else {
+      return getPictureUtility().downloadPicture(url);
+    }
+  }
+
+  // Save/load pictures from local storage
+  @Override
+  public void savePictureToLocal(Context context, Favor favor, Bitmap picture) {
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:178]: make private

File: app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java [hunk 4]
```diff
@@ -1,253 +1,282 @@
...
+//    String favorId = favor.getId();
+//    String imageNum = "0"; // Todo: Support multiple images
+//    CacheUtil.SaveToStorageParams saveToStorageParams = new CacheUtil.SaveToStorageParams(
+//            baseDir, favorId, imageNum, picture
+//    );
+//    saveToStorageTask.execute(saveToStorageParams);
+  }
+
+  @Override
+  public CompletableFuture<Bitmap> loadPictureFromLocal(Context context, Favor favor) {
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:191]: Should make private

File: app/src/main/java/ch/epfl/favo/viewmodel/IFavorViewModel.java [hunk 5]
```diff
@@ -1,46 +1,52 @@
-package ch.epfl.favo.viewmodel;
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/viewmodel/IFavorViewModel.java:1]: Are you reformatting the code? For some reason it doesn't track changes properly

Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/viewmodel/IFavorViewModel.java:1]: No, I don't think I touched this.

File: app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java (hunk 1 above)
Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java:151]: Yeah let's look at that in the next sprint. I also need to include this in FavorDetailView.

File: app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java (hunk 2 above)
Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:163]: What to make private?

Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:163]: Hm but I'm implementing the interface

File: app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java (hunk 3 above)
Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:178]: Same as above

File: app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java (hunk 4 above)
Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java:191]: Same as above

**Pull request merged**